spec/unit/hint_compiler_spec.rb

```diff
@@ -23,14 +23,14 @@
       ],
       [
         :and, [
-          [:val, [:predicate, [:attr?, [:height]]]],
+          [:val, [:predicate, [:key?, [:height]]]],
           [
             :or, [
               [:attr, [:height, [:predicate, [:none?, []]]]],
               [
                 :and, [
-                  [:attr, [:height, [:predicate, [:int?, []]]]],
-                  [:attr, [:height, [:predicate, [:gt?, [180]]]]]
+                  [:key, [:height, [:predicate, [:int?, []]]]],
+                  [:key, [:height, [:predicate, [:gt?, [180]]]]]
```

Comment:
The tests passed without making this change but I believe that is because dry-logic supports `attr?` as a predicate - I updated the tests to reflect the supported use cases.